internal/apis/certmanager/validation/certificate.go

```diff
@@ -54,11 +54,25 @@ func ValidateCertificateSpec(crt *internalcmapi.CertificateSpec, fldPath *field.
 
 	var commonName = crt.CommonName
 	if crt.LiteralSubject != "" {
-
 		if !utilfeature.DefaultFeatureGate.Enabled(feature.LiteralCertificateSubject) {
 			el = append(el, field.Forbidden(fldPath.Child("literalSubject"), "Feature gate LiteralCertificateSubject must be enabled on both webhook and controller to use the alpha `literalSubject` field"))
 		}
 
+		if len(crt.CommonName) != 0 {
+			el = append(el, field.Invalid(fldPath.Child("commonName"), crt.CommonName, "When providing a `LiteralSubject` no `commonName` may be provided."))
+		}
+
+		if crt.Subject != nil && (len(crt.Subject.Organizations) > 0 ||
+			len(crt.Subject.Countries) > 0 ||
+			len(crt.Subject.OrganizationalUnits) > 0 ||
+			len(crt.Subject.Localities) > 0 ||
+			len(crt.Subject.Provinces) > 0 ||
+			len(crt.Subject.StreetAddresses) > 0 ||
+			len(crt.Subject.PostalCodes) > 0 ||
+			len(crt.Subject.SerialNumber) > 0) {
```

Comment:
Currently we allow `SerialNumber` to be defined in the subject when `LiteralSubject` is defined. There is no reason this should be allowed.
The serial number can just be set using the `LiteralSubject` field.

Reply:
I think in https://github.com/cert-manager/cert-manager/pull/5002, we might have confused the Certificate SerialNumber with the SerialNumber in the Subject (see https://datatracker.ietf.org/doc/html/rfc5280#section-4.1.2.4).
<img width="385" alt="image" src="https://github.com/cert-manager/cert-manager/assets/42113979/24397e21-9612-4b97-b750-ad383a8539ff">

Reply:
Yeah that's my bad - the serial number is the subject one and not the cert one. I'm surprised that we have the ability to set the subject serial number but not the certificate serial number. I'll amend my comments in that PR to avoid spreading any more confusion on that!